test/instrumentation/modules/undici/undici.test.js

```diff
@@ -240,15 +242,32 @@ if (global.AbortController) {
         { name: 'aTransName', type: 'manual', sampled: true },
         'error.transaction',
       );
-      t.equal(
-        error.exception.message,
-        'Request aborted',
-        'error.exception.message',
-      );
-      t.equal(error.exception.type, 'AbortError', 'error.exception.type');
-      t.equal(error.exception.code, 'UND_ERR_ABORTED', 'error.exception.code');
-      t.equal(error.exception.module, 'undici', 'error.exception.module');
       t.equal(error.exception.handled, true, 'error.exception.handled');
+      t.equal(error.exception.type, 'AbortError', 'error.exception.type');
+      if (semver.gte(undiciVer, '6.3.0')) {
+        // In undici@6.3.0 (https://github.com/nodejs/undici/pull/2592)
+        // `abort.reason` is used, if any. Node.js core AbortController's
+        // default reason uses "This operation was aborted".
+        t.equal(
+          error.exception.message,
+          'This operation was aborted',
+          'error.exception.message',
+        );
+        // https://developer.mozilla.org/en-US/docs/Web/API/DOMException#aborterror
+        t.equal(error.exception.code, '20', 'error.exception.code');
```

Comment:
if the lib instrumented changes the error codes and messages I guess it's expected and anyone having a query on ElasticSearch/Kibana should update it accordingly. But shouldn't `error.exception.module` be defined here with the value `undici`?

Reply:
IIRC `error.exception.module` comes from reading the `err.stack`. Now that Undici is passing through the Error object set on the `abort.reason` instead of creating its own, the `err.stack` points to a location in Node.js core where it is created (this loc: https://github.com/nodejs/node/blob/e133e5115a829aa7a445cfd8754df3a3dd586b0a/lib/internal/abort_controller.js#L395). That means the top frame in `err.stack` no longer points to a location in the `undici` module.